crates/phala-types/src/contract.rs

```diff
@@ -26,6 +26,11 @@ pub enum CodeIndex<CodeHash> {
     WasmCode(CodeHash),
 }
 
+#[derive(Decode, Encode, TypeInfo)]
+pub enum InkCommand {
+    InkMessage { nonce: Vec<u8>, message: Vec<u8> },
```

Comment:
What's the functionality of `nonce` here? Should we maintain it in every pallet that sends messages to ink contract?

Reply:
The nonce doesn't affect the contract execution itself. But the cluster owner can config a log server to receive the ink message execution result. pRuntime would send the result with the nonce to the log server. Consider it a dev tool for contract development.

Reply:
I am not sure if there are potential issues that exist, suppose there are multiple pallets sending messages to FatContract, unless MQ can bind the `nonce` with the sending pallet, we may get race condition of `nonce`. And if no race condition exists, we just simply increase `nonce` one by one right?

Reply:
It is not necessary to be unique. You can leave it empty as this PR does.

Reply:
Cool, LGTM then